java-vtl-script/src/main/java/no/ssb/vtl/script/functions/VTLSubstr.java

```diff
@@ -0,0 +1,98 @@
+package no.ssb.vtl.script.functions;
+
+/*
+ * ========================LICENSE_START=================================
+ * Java VTL
+ * %%
+ * Copyright (C) 2016 - 2017 Pawel Buczek
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =========================LICENSE_END==================================
+ */
+
+import no.ssb.vtl.model.VTLInteger;
+import no.ssb.vtl.model.VTLString;
+
+import static java.lang.String.*;
+
+public class VTLSubstr extends AbstractVTLFunction<VTLString> {
+
+    private static final String ARGUMENT_GREATER_THAT_ZERO = "%s must be greater than zero, was %s";
+    private static final Argument<VTLString> DS = new Argument<>("ds", VTLString.class);
+    private static final Argument<VTLInteger> START_POSITION = new Argument<>("startPosition", VTLInteger.class);
+    private static final Argument<VTLInteger> LENGTH = new Argument<>("length", VTLInteger.class);
+    private static VTLSubstr instance;
+    private VTLString STRING_NULL = VTLString.of((String) null);
+    private VTLInteger INTEGER_NULL = VTLInteger.of((Long) null);
+
+    private VTLSubstr() {
+        super("substr", VTLString.class, DS, START_POSITION, LENGTH);
+    }
+
+    public static VTLSubstr getInstance() {
+        if (instance == null)
+            instance = new VTLSubstr();
+        return instance;
+    }
+
+    @Override
+    protected VTLString safeInvoke(TypeSafeArguments arguments) {
+        VTLString value = arguments.getNullable(DS, STRING_NULL);
+        VTLInteger startPosition = arguments.getNullable(START_POSITION, INTEGER_NULL);
+        VTLInteger length = arguments.getNullable(LENGTH, INTEGER_NULL);
+
+        if (value.get() == null) {
+            return VTLString.of((String) null);
+        }
+
+        if (value.get().isEmpty()) {
+            return VTLString.of("");
```

Comment:
How do you interpret this? Every string parameter can be converted to `VTLString.of("")`, if so we should find a way to avoid repeating this check everywhere.

Reply:
Yes, I think we should handle this in one central place.
I think it means we need to treat `""` (empty strings) as if it is `null` in boolean operations and as if the is a string value in string operations

Reply:
We discussed this with Eivind, we could change VTLString.of to that is returns null if the string is empty: 
```
    public static VTLString of(String string) {
        return new VTLString() {
            /**
             * Returns the value of the data point.
             */
            @Override
            public String get() {
                return "".equals(string) ? null : string;
            }
        };
    }
``` 
Another way could be to implement a isNull() method in the VTLObject interface that we use to check null on VTL types. It is anyway not a good idea to have this in the function itself.